src/app/public/modules/back-to-top/back-to-top-adapter.service.ts

```diff
@@ -19,23 +26,67 @@ import {
   takeUntil
 } from 'rxjs/operators';
 
+import {
+  SkyBackToTopComponent
+} from './back-to-top.component';
+
+import {
+  SkyBackToTopType
+} from './models/back-to-top-type';
+
 /**
  * @internal
  */
 @Injectable()
 export class SkyBackToTopDomAdapterService implements OnDestroy {
 
   private ngUnsubscribe = new Subject<void>();
+  private renderer: Renderer2;
 
   constructor(
-    private windowRef: SkyAppWindowRef
-  ) { }
+    private componentFactoryResolver: ComponentFactoryResolver,
+    private applicationRef: ApplicationRef,
+    private injector: Injector,
+    private windowRef: SkyAppWindowRef,
+    private rendererFactory: RendererFactory2
+  ) {
+    // Based on suggestions from https://github.com/angular/angular/issues/17824
+    // for accessing an instance of Renderer2 in a service since Renderer2 can't
+    // be injected into a service.  Passing undefined for both parameters results
+    // in the default renderer which is what we want here.
+    this.renderer = this.rendererFactory.createRenderer(undefined, undefined);
+  }
 
   public ngOnDestroy(): void {
     this.ngUnsubscribe.next();
     this.ngUnsubscribe.complete();
   }
 
+  public createComponentWithinModal(elementRef: ElementRef): ComponentRef<SkyBackToTopComponent> {
+    const componentRef = this.componentFactoryResolver
+    .resolveComponentFactory<SkyBackToTopComponent>(SkyBackToTopComponent)
+    .create(this.injector);
+
+    this.applicationRef.attachView(componentRef.hostView);
+
+    const modal = this.findModalElement(elementRef.nativeElement);
+    const modalFooter = modal.querySelector('.sky-modal-footer');
```

Comment:
Looking for a specific CSS class worries me a bit. We're essentially looking for a parent DOM element and branching on it, which couples the logic to not only a particular component, but the internal DOM structure of that component. I would suggest using Angular's dependency injection hierarchy to solve this, where a parent element provides context to the child element on how to behave. A good example of this pattern is the flyout component, where it provides an alternate implementation of `SkyMediaQueryService` so that child components will respond down to the dimensions of the flyout instead of the dimensions of the page. Perhaps something like a `SkyBackToTopHostProvider` wo